lib/fastlane/plugin/firebase_app_distribution/client/firebase_app_distribution_api_client.rb

```diff
@@ -107,25 +107,27 @@ def upload_binary(app_id, binary_path)
       #
       # Returns the release_id on a successful release, otherwise returns nil.
       #
-      # Throws an error if the number of polling retries exceeds MAX_POLLING_RETRIES
+      # Throws a UI error if the number of polling retries exceeds MAX_POLLING_RETRIES
+      # or if it was not able to upload the binary
       def upload(app_id, binary_path)
         upload_token = get_upload_token(app_id, binary_path)
         upload_status_response = get_upload_status(app_id, upload_token)
         if upload_status_response.success?
           UI.success("This APK/IPA has been uploaded before. Skipping upload step.")
         else
-          UI.message("This APK has not been uploaded before.")
+          UI.message("Uploading the APK/IPA.")
+          upload_binary(app_id, binary_path)
```

Comment:
The one case I see this logic having issues is if the upload times out and they try again, and the previous upload is still in progress. In that case, we would try to upload again here, and we would get an error from the server because the binary still exists (I assume) and crash.

I think ideally in that scenario we would skip the second upload and enter the loop to keep checking the status.

Reply:
+1 good catch Lee!